aws/data_source_aws_kms_secrets.go

```diff
@@ -83,6 +93,16 @@ func dataSourceAwsKmsSecretsRead(d *schema.ResourceData, meta interface{}) error
 			}
 		}
 
+		key_id := secret["key_id"]
+		if key_id != "" {
+			params.KeyId = aws.String(key_id.(string))
+		}
+
+		encryption_algorithm := secret["encryption_algorithm"]
+		if encryption_algorithm != "" {
+			params.EncryptionAlgorithm = aws.String(encryption_algorithm.(string))
+		}
```

Comment:
`if key_id, exists := secret["key_id"]; exists {}` approach does not really work, becasue `key_id` is apparently an emptry string, which results in `KeyId` bening added to the `params` and AWS API returning an error since it [expects](https://pkg.go.dev/github.com/aws/aws-sdk-go/service/kms#DecryptInput) at least 1 input for `KeyId`.

So to not include an empty `KeyId` in the `params`, I'm checking for `key_id != ""`.

Reply:
```
log.Printf("%s", params)

2021-06-10T16:48:12.392+0200 [INFO]  plugin.terraform-provider-aws: 2021/06/10 16:48:12 {
  CiphertextBlob: <binary> len 512,
  EncryptionContext: {

  },
  GrantTokens: [],
  KeyId: ""
}
```